ipinfo/config.go

```diff
@@ -32,74 +32,127 @@ func getConfigDir() (string, error) {
 	return confDir, nil
 }
 
-// init function
+// returns the path to the config file.
+func ConfigPath() (string, error) {
+	confDir, err := getConfigDir()
+	if err != nil {
+		return "", err
+	}
+
+	return filepath.Join(confDir, "config.json"), nil
+}
+
+// returns the path to the token file.
+func TokenPath() (string, error) {
+	confDir, err := getConfigDir()
+	if err != nil {
+		return "", err
+	}
+
+	return filepath.Join(confDir, "token"), nil
+}
+
 func InitConfig() error {
-	path, err := ConfigPath()
+	configpath, err := ConfigPath()
+	if err != nil {
+		return err
+	}
+
+	tokenpath, err := TokenPath()
 	if err != nil {
 		return err
 	}
 
-	if !lib.FileExists(path) {
-		err := SetConfig(NewConfig())
+	// create default config if none yet.
+	if !lib.FileExists(configpath) {
+		gConfig = NewConfig()
+	} else {
+		config, err := ReadConfig()
 		if err != nil {
 			return err
 		}
+		gConfig = config
 	}
-	return nil
-}
 
-// returns the path to config file.
-func ConfigPath() (string, error) {
-	confDir, err := getConfigDir()
-	if err != nil {
-		return "", err
+	// if token exists, migrate it to config file.
+	if lib.FileExists(tokenpath) {
+		token, err := TokentoConfig()
+		if err != nil {
+			return err
+		}
+		gConfig.Token = token
+
+		// remove the existing token file
+		if err := os.Remove(tokenpath); err != nil {
+			return err
+		}
 	}
-	return filepath.Join(confDir, "config.json"), nil
+	if err := SaveConfig(gConfig); err != nil {
+		return err
+	}
+
+	return nil
 }
 
-// returns config file with default settings.
+// returns a new, default config.
 func NewConfig() Config {
 	return Config{
-		Cache: true,
-		Token: "",
+		CacheEnabled: true,
+		Token:        "",
 	}
 }
 
-// set the values of config.
-func SetConfig(config Config) error {
+// migration of token to config file.
+//
+// might be deleted in future release.
+func TokentoConfig() (string, error) {
+	path, err := TokenPath()
+	if err != nil {
+		return "", err
+	}
+	token, err := ioutil.ReadFile(path)
+	if err != nil {
+		return "", err
+	}
+
+	return string(token), nil
+}
+
+// save the config to file.
+func SaveConfig(config Config) error {
 	configPath, err := ConfigPath()
 	if err != nil {
 		return err
 	}
 
-	jsonFile, err := json.Marshal(config)
+	jsonData, err := json.Marshal(config)
 	if err != nil {
 		return err
 	}
 
-	err = ioutil.WriteFile(configPath, jsonFile, 0644)
-	if err != nil {
+	if err = ioutil.WriteFile(configPath, jsonData, 0644); err != nil {
```

Comment:
This is not a huge deal but what it does is, if we had a variable named `err` in this function scope, doing `err :=` here will not override the value of that higher-scope `err`. This is not too important right here but better practice to use `:=` unless you explicitly intend on overriding the value of the higher-scoped `err` variable.

```suggestion
	if err := ioutil.WriteFile(configPath, jsonData, 0644); err != nil {
```